### PR TITLE
[Enhancement] Added allocate payment amount checkbox in Payment Entry

### PR DESCRIPTION
**Pull request by rohitwaghchaure**

### CODE REVIEW

File: erpnext/accounts/doctype/payment_entry/payment_entry.js [hunk 1]
```diff
@@ -511,6 +511,17 @@ frappe.ui.form.on('Payment Entry', {
 		});
 	},
 
+	allocate_payment_amount: function(frm) {
```
Comment (nabinhait) [inline erpnext/accounts/doctype/payment_entry/payment_entry.js]: if payment type is internal transfer, then return

File: erpnext/accounts/doctype/payment_entry/payment_entry.js [hunk 2]
```diff
@@ -511,6 +511,17 @@ frappe.ui.form.on('Payment Entry', {
...
+		if(frm.doc.references.length == 0){
+			frm.events.get_outstanding_documents(frm);
+		}
+
+		if(frm.doc.payment_type == "Pay")
+			frm.events.allocate_party_amount_against_ref_docs(frm, frm.doc.received_amount);
+		else
```
Comment (nabinhait) [inline erpnext/accounts/doctype/payment_entry/payment_entry.js]: if..else is not required, as we are returning if payment type is Internal Transfer

File: erpnext/accounts/doctype/payment_entry/payment_entry.json [hunk 3]
```diff
@@ -777,6 +777,33 @@
    "bold": 0, 
    "collapsible": 0, 
    "columns": 0, 
+   "default": "1", 
+   "fieldname": "allocate_payment_amount",
```
Comment (nabinhait) [inline erpnext/accounts/doctype/payment_entry/payment_entry.json]: Set depends on `eval:in_list(['Pay', 'Receive'], doc.payment_type)`

File: erpnext/accounts/doctype/payment_entry/payment_entry.js (hunk 2 above)
Comment (nabinhait) [inline erpnext/accounts/doctype/payment_entry/payment_entry.js]: if allocate checkbox is unchecked, set payment amount against all ref doc to zero.

**Review by nabinhait — CHANGES_REQUESTED** [review]

File: erpnext/accounts/doctype/payment_entry/payment_entry.json (hunk 3 above)
Comment (rohitwaghchaure) [inline erpnext/accounts/doctype/payment_entry/payment_entry.json]: Done

File: erpnext/accounts/doctype/payment_entry/payment_entry.js (hunk 2 above)
Comment (rohitwaghchaure) [inline erpnext/accounts/doctype/payment_entry/payment_entry.js]: Added return statement if payment type is internal transfer.

**Pull request merged**